Unit test cases for hp network provisioning driver

selvakumars2 opened the pull request:
This patchset implemented the back-end mechanism driver unit test cases

hp-networking commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,128 @@
+from neutron.tests import base
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
```
remove log files from UT.

hp-networking commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,128 @@
+            self.assertRaises(hp_exec.ConnectionFailed,
+                              self.driver.bind_port_to_segment,
+                              port_dict)
+
```
Can you please add a testcase for bind_ignore use-case

hp-networking commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,128 @@
+                                   return_value=None):
+                with mock.patch.object(db, 'add_hp_ironic_switch_port_mapping',
+                                       return_value=None):
+                    self.driver.create_port(port_dict)
```
Add a test cases for 200 OK but switch not known, if possible.

hp-networking commented:
Comments inline.

selvakumars2 commented:
Hi Vivek,
Thanks for reviewing the unit test cases,
I have posted another patchset with below changes included,
1. removed the log statement as per your suggesstion
2. if the switch is not known , added a test case for that
3. bind failure cases handled
4. bind success we need to handle in front-end driver

let me know what you think on this

koteswar16 commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,148 @@
+from baremetal_network_provisioning.common import exceptions as hp_exec
+from baremetal_network_provisioning.db import bm_nw_provision_db as db
+from baremetal_network_provisioning.db import bm_nw_provision_models as models
+from baremetal_network_provisioning.ml2 import hp_network_provisioning_driver as driver
```
>79 chars

koteswar16 commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,148 @@
+            with mock.patch.object(db, 'add_hp_switch_port',
+                                   return_value=None):
+                with mock.patch.object(db, 'add_hp_ironic_switch_port_mapping',
+                                       return_value=None):
```
can we use contextlib.nested here?

hp-networking replied:
Please address this comment from koteswar in the next new review.

koteswar16 commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,148 @@
+                    self.driver.create_port(port_dict)
+
+    def test_create_port_with_connection_failed(self):
+        """test_create_port_with_connection_failed."""
```
docstring should be some description about the method.
"""" Test create_port_with_connection_failed method."""

koteswar16 commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,148 @@
+                                      port_dict)
+            self.assertEqual(' Connection has failed: 404 Client Error: None',
+                             error.msg)
+
```
add unit test for SSLError.

koteswar16 replied:
add unit test which covers hp_exec.HPNetProvisioningDriverError

koteswar16 commented on baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py:
```diff
@@ -0,0 +1,148 @@
+#    under the License.
+#
+from baremetal_network_provisioning.common import constants as hp_const
+from baremetal_network_provisioning.common import exceptions as hp_exec
```
name hp_exec is misleading. it can be hp_exc

selvakumars2 commented:
Addressed Koteswara's comments.
